carapace-server/src/main/java/org/carapaceproxy/api/CertificatesResource.java

```diff
@@ -165,82 +172,54 @@ private CertificateBean findCertificateById(String certId) {
             CertificateBean certBean = new CertificateBean(
                     certificate.getId(),
                     certificate.getHostname(),
-                    stateToStatusString(certificate.getMode()),
+                    certificateModeToString(certificate.getMode()),
                     certificate.isDynamic(),
                     certificate.getFile()
             );
 
             if (certificate.isDynamic()) {
                 DynamicCertificateState state = server.getDynamicCertificateManager().getStateOfCertificate(certBean.getId());
-                certBean.setStatus(stateToStatusString(state));
+                certBean.setStatus(certificateStateToString(state));
             }
             return certBean;
         }
 
         return null;
     }
 
-    static String stateToStatusString(DynamicCertificateState state) {
-        if (state == null) {
-            return "unknown";
-        }
-        switch (state) {
-            case WAITING:
-                return "waiting"; // certificate waiting for issuing/renews
-            case VERIFYING:
-                return "verifying"; // challenge verification by LE pending
-            case VERIFIED:
-                return "verified"; // challenge succeded
-            case ORDERING:
-                return "ordering"; // certificate order pending
-            case REQUEST_FAILED:
-                return "request failed"; // challenge/order failed
-            case AVAILABLE:
-                return "available";// certificate available(saved) and not expired
-            case EXPIRED:     // certificate expired
-                return "expired";
-            default:
-                return "unknown";
-        }
-    }
-
-    static String stateToStatusString(CertificateMode mode) {
-        if (mode == null) {
-            return "unknown";
-        }
-        switch (mode) {
-            case STATIC:
-                return "static";
-            case ACME:
-                return "acme";
-            case MANUAL:
-                return "manual";
-            default:
-                return "unknown";
-        }
-    }
-
     @POST
     @Path("{domain}/upload")
     @Consumes(MediaType.APPLICATION_OCTET_STREAM)
-    public Response uploadCertificate(@PathParam("domain") String domain, InputStream uploadedInputStream) throws Exception {
+    public Response uploadCertificate(
+            @PathParam("domain") String domain,
+            @QueryParam("type") @DefaultValue("manual") String type,
+            InputStream uploadedInputStream) throws Exception {
 
+        // Certificate type (manual | acme)
+        CertificateMode certType = stringToCertificateMode(type);
+        if (certType == null || STATIC.equals(certType)) {
+            return Response.status(422).entity("ERROR: illegal type of certificate. Available: manual, acme").build();
+        }
+        // Certificate content (optional)
         byte[] data = uploadedInputStream.readAllBytes();
```

Comment:
we aren't closing this inputstream, what is the JAX-RS contract with this stream ?